homeassistant/components/iotawatt/sensor.py

```diff
@@ -0,0 +1,168 @@
+"""Support for IoTaWatt Energy monitor."""
+from __future__ import annotations
+
+from iotawattpy.sensor import Sensor
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.const import (
+    DEVICE_CLASS_CURRENT,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_POWER_FACTOR,
+    DEVICE_CLASS_VOLTAGE,
+    ELECTRIC_CURRENT_AMPERE,
+    ELECTRIC_POTENTIAL_VOLT,
+    ENERGY_WATT_HOUR,
+    FREQUENCY_HERTZ,
+    PERCENTAGE,
+    POWER_VOLT_AMPERE,
+    POWER_WATT,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import entity_registry, update_coordinator
+
+from .const import DOMAIN, VOLT_AMPERE_REACTIVE, VOLT_AMPERE_REACTIVE_HOURS
+from .coordinator import IotawattUpdater
+
+ENTITY_DESCRIPTION_KEY_MAP: dict[str, SensorEntityDescription] = {
+    "Amps": SensorEntityDescription(
+        "Amps",
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        device_class=DEVICE_CLASS_CURRENT,
+    ),
+    "Hz": SensorEntityDescription(
+        "Hz",
+        native_unit_of_measurement=FREQUENCY_HERTZ,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    "PF": SensorEntityDescription(
+        "PF",
+        native_unit_of_measurement=PERCENTAGE,
```

Comment:
Will we need a conversion now from +- 1 to +- 100 %? If so, we can add a lambda function in the entity descriptions for formatting conversions.

Reply:
The current value returned by the IoTaWatt device appears to be a value between 0 and 1, inclusive. Would a function be required to convert the value to a percentage, for example ```0.93``` to ```93```?

Reply:
Yes. Please add an optional lambda and define a custom sensor entity description for this integration.

Example:
https://github.com/home-assistant/core/blob/de5a22953d22fc22e3fa7a2656b3a72050c91dc0/homeassistant/components/airly/sensor.py#L61-L65

https://github.com/home-assistant/core/blob/de5a22953d22fc22e3fa7a2656b3a72050c91dc0/homeassistant/components/airly/sensor.py#L96-L103

```py
@dataclass 
class IotaWattSensorEntityDescription(SensorEntityDescription): 
    """Class describing IotaWatt sensor entities.""" 

    value: Callable | None = None

Reply:
I added the new class ```IotaWattSensorEntityDescription```, added the lambda function, and updated ```native_value()``` function to use the new value. Thanks @balloob! 🎉